python/src/pyunigen.cpp

```diff
@@ -321,7 +328,15 @@ static PyObject* sample(Sampler *self, PyObject *args, PyObject *kwds)
     }
 
     self->appmc->set_projection_set(self->sampling_set);
-    auto sol_count = self->appmc->count();
+    
+    auto sol_count = * new ApproxMC::SolCount;
```

Comment:
Don't create a `new` here. Just do
```
ApproxMC::SolCount solcount;
```

Then overwrite it if need be. You never free sol_count, for example. No need for dynamic allocation here.